libmamba/src/api/install.cpp

```diff
@@ -1010,5 +1010,182 @@ namespace mamba
                 }
             }
         }
+
+        // The following function parses the different formats that can be found in the history
+        // file. conda/mamba1 format: installed: +conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        // removed: -conda-forge/linux-64::xtl-0.8.0-h84d6215_0 mamba2 broken format: installed:
+        // +conda-forge::xtl-0.8.0-h84d6215_0 removed:
+        // -https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0 mamba2 new format:
+        // installed: +https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        // removed: -https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        specs::PackageInfo pkg_info_builder(std::string s)
+        {
+            std::string s_pkg;
+            std::string channel;
+            size_t pos_0 = s.rfind("/");
+            if (pos_0 != std::string::npos)
+            {
+                std::string s_begin = s.substr(0, pos_0);
+                std::string s_end = s.substr(pos_0 + 1, s.size());
+                size_t pos = s_begin.rfind("/");
+                if (pos != std::string::npos)
+                {
+                    channel = s_begin.substr(pos + 1, s_begin.size());
+                }
+                else
+                {
+                    channel = s_begin;
+                }
+                s_pkg = util::split(s_end, "::").back();
```

Comment:
`util::split`, splits everything and creates a temporary vector, this is more efficient:
```suggestion
                s_pkg = std::get<1>(util::rsplit_once(s_end, "::"));
```